frontend/src/actions/TableActions.js

```diff
@@ -511,8 +512,10 @@ function updateCollapsedRows({ tableId, rows }) {
       collapsedRows,
       collapsedParentRows,
     } = table;
-    let newCollapsedParentRows = [];
-    let newCollapsedRows = [];
+
+    const hasChangedIds = changedIds && changedIds.length > 0;
+    let newCollapsedParentRows = hasChangedIds ? [...collapsedParentRows] : [];
+    let newCollapsedRows = hasChangedIds ? [...collapsedRows] : [];
```

Comment:
unfortunately, I am not able to review this code so we will have to rely on
* @siemiatj  if u have an idea?
* documented tests you performed